internal/controller/common.go

```diff
@@ -482,6 +482,8 @@ func (r *PackageReconcilationContext) ensureDependencies(ctx context.Context) bo
 		}
 	}
 
+	ownerutils.Add(&r.currentOwnedPackages, ownedPackages...)
```

Comment:
I think we should also call `r.setShouldUpdate` here 

```suggestion
	r.setShouldUpdate(ownerutils.Add(&r.currentOwnedPackages, ownedPackages...))
```